servicetalk-http-api/src/main/java/io/servicetalk/http/api/HttpClientBuilder.java

```diff
@@ -43,37 +45,126 @@
+     */
+    public abstract HttpClientBuilder<U, R, SDE> disableHostHeaderFallback();
 
-    @Override
-    public abstract HttpClientBuilder<U, R, SDE> enableWireLogging(String loggerName,
-                                                                   LogLevel logLevel,
-                                                                   BooleanSupplier logUserData);
+    /**
+     * Provide a hint if response <a href="https://tools.ietf.org/html/rfc7230#section-4.1.2">trailers</a> are allowed
+     * to be dropped. This hint maybe ignored if the transport can otherwise infer that
+     * <a href="https://tools.ietf.org/html/rfc7230#section-4.1.2">trailers</a> should be preserved. For example, if the
+     * response headers contain <a href="https://tools.ietf.org/html/rfc7230#section-4.4">Trailer</a> then this hint
+     * maybe ignored.
+     * @param allowDrop {@code true} if response
+     * <a href="https://tools.ietf.org/html/rfc7230#section-4.1.2">trailers</a> are allowed to be dropped.
+     * @return {@code this}
+     */
+    public abstract HttpClientBuilder<U, R, SDE> allowDropResponseTrailers(boolean allowDrop);
+
+    /**
+     * Appends the filter to the chain of filters used to decorate the {@link StreamingHttpConnection} created by this
+     * builder.
+     * <p>
+     * Filtering allows you to wrap a {@link StreamingHttpConnection} and modify behavior during request/response
+     * processing
+     * Some potential candidates for filtering include logging, metrics, and decorating responses.
+     * <p>
+     * The order of execution of these filters are in order of append. If 3 filters are added as follows:
+     * <pre>
+     *     builder.append(filter1).append(filter2).append(filter3)
+     * </pre>
+     * making a request to a connection wrapped by this filter chain the order of invocation of these filters will be:
+     * <pre>
+     *     filter1 ⇒ filter2 ⇒ filter3 ⇒ connection
+     * </pre>
+     * @param factory {@link StreamingHttpConnectionFilterFactory} to decorate a {@link StreamingHttpConnection} for the
+     * purpose of filtering.
+     * @return {@code this}
+     */
+    public abstract HttpClientBuilder<U, R, SDE> appendConnectionFilter(
+            StreamingHttpConnectionFilterFactory factory);
+
+    /**
+     * Appends the filter to the chain of filters used to decorate the {@link StreamingHttpConnection} created by this
+     * builder, for every request that passes the provided {@link Predicate}.
+     * <p>
+     * Filtering allows you to wrap a {@link StreamingHttpConnection} and modify behavior during request/response
+     * processing
+     * Some potential candidates for filtering include logging, metrics, and decorating responses.
+     * <p>
+     * The order of execution of these filters are in order of append. If 3 filters are added as follows:
+     * <pre>
+     *     builder.append(filter1).append(filter2).append(filter3)
```

Comment:
We recently improved javadoc for `append*`methods in #1710 and #1729. While you are here, can you please use the actual method name in these cases? For example, here it will be:
```
* builder.appendConnectionFilter(filter1).appendConnectionFilter(filter2).appendConnectionFilter(filter3)
```

Reply:
I understand you refer to the documentation of the server filters, as I haven't noticed any changes in the client filters in the PRs you linked. I will change the name as requested, but I'm not sure how to proceed about the referred PRs, please let me know.